src/lib/rest/StringFilter.cpp

```diff
@@ -1255,7 +1328,20 @@ bool StringFilter::mongoFilterPopulate(std::string* errorStringP)
       left = itemP->attributeName;
     }
 
-    if (left == DATE_CREATED)
+ 
+    //
+    // Also, if the left hand side contains a compound path, then we are
+    // dealing with matching of a compound item.
+    //
+    if (itemP->compoundPath.size() != 0)
+    {
+      if (itemP->type == SftQ)
+      {
+        left = std::string(ENT_ATTRS) + "." + itemP->attributeName + ".value." + itemP->compoundPath;
```

Comment:
If this is going to be a filter for the DB, then ".value." should be "." + ENT_ATTRS_VALUE + "."

(I'm writting this without checking code... pls, check that ENT_ATTRS_VALUE is the right name).